File: app-kotlin/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/data/viewmodels/shared/OnboardingViewModel.java
package com.TritiumGaming.phasmophobiaevidencepicker.data.viewmodels.shared;

import android.content.Context;
import android.content.SharedPreferences;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import androidx.lifecycle.ViewModel;

import com.TritiumGaming.phasmophobiaevidencepicker.R;

import java.util.HashMap;

/** @noinspection SameParameterValue*/
public class OnboardingViewModel extends SharedViewModel {

    private boolean canShowIntroduction = true;

    @Override
    public void setFileName() {
        fileName = R.string.preferences_onboardingFile_name;
    }

    public boolean init(@NonNull Context context) {

        setFileName();

        SharedPreferences sharedPref = getSharedPreferences(context);

        setCanShowIntroduction(sharedPref.getBoolean(context.getResources().getString(R.string.onboarding_canShow_intro), getCanShowIntroduction()));

        saveToFile(context);

        return true;
    }

    public void setCanShowIntroduction(boolean canShowIntroduction) {
        this.canShowIntroduction = canShowIntroduction;
    }

    public boolean getCanShowIntroduction() {
        return canShowIntroduction;
    }
    /**
     *
     * @param c
     * @param editor
     * @param localApply
     */
    public void saveCanShowIntroduction(
            @NonNull Context c, @Nullable SharedPreferences.Editor editor, boolean localApply) {
        if(editor == null) {
            editor = getEditor(c);
        }

        editor.putBoolean(c.getResources().getString(R.string.tutorialTracking_canShowIntroduction),
                getCanShowIntroduction());

        if(localApply) {
            editor.apply();
        }
    }

    /**
     * saveToFile method
     *
     * @param context The Activity context.
     */
    public void saveToFile(@NonNull Context context) {

        SharedPreferences.Editor editor = getEditor(context);

        saveCanShowIntroduction(context, editor, false);

        editor.apply();
    }

    @NonNull
    public HashMap<String, String> getDataAsList() {
        HashMap<String, String> settings = new HashMap<>();
        settings.put("can_show_intro", String.valueOf(getCanShowIntroduction()));

        return settings;
    }

    /**
     *
     * @param context
     */
    public void printFromFile(@NonNull Context context) {
        SharedPreferences sharedPref = getSharedPreferences(context);

        Log.d("OnboardingPrefs",
                "Can Show Introduction: " + sharedPref.getBoolean(context.getResources().getString(R.string.tutorialTracking_canShowIntroduction), false));
    }

    /**
     *
     */
    public void printFromVariables() {
        Log.d("GlobalPreferencesVars",
                "; Can Show Introduction: " + getCanShowIntroduction());
    }

}

File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/data/viewmodels/shared/OnboardingViewModel.kt
package com.TritiumGaming.phasmophobiaevidencepicker.data.viewmodels.shared

import android.content.Context
import android.content.SharedPreferences
import android.util.Log
import com.TritiumGaming.phasmophobiaevidencepicker.R

/** @noinspection SameParameterValue
 */
class OnboardingViewModel : SharedViewModel() {
    @JvmField
    var canShowIntroduction: Boolean = true

    override fun setFileName() {
        fileName = R.string.preferences_onboardingFile_name
    }

    override fun init(context: Context): Boolean {
        setFileName()

        val sharedPref = getSharedPreferences(context)

        canShowIntroduction = sharedPref.getBoolean(
            context.resources.getString(R.string.onboarding_canShow_intro),
            canShowIntroduction
        )

        saveToFile(context)

        return true
    }

    /**
     *
     * @param c
     * @param editor
     * @param localApply
     */
    fun saveCanShowIntroduction(
        c: Context, editor: SharedPreferences.Editor?, localApply: Boolean
    ) {
        var editor = editor
        if (editor == null && (getEditor(c).also { editor = it }) == null) {
            return
        }

        editor!!.putBoolean(
            c.resources.getString(R.string.tutorialTracking_canShowIntroduction),
            canShowIntroduction
        )

        if (localApply) {
            editor!!.apply()
        }
    }

    /**
     * saveToFile method
     *
     * @param context The Activity context.
     */
    override fun saveToFile(context: Context) {
        val editor = getEditor(context)

        saveCanShowIntroduction(context, editor, false)

        editor!!.apply()
    }

    val dataAsList: HashMap<String, String>
        get() {
            val settings = HashMap<String, String>()
            settings["can_show_intro"] = canShowIntroduction.toString()

            return settings
        }

    /**
     *
     * @param context
     */
    fun printFromFile(context: Context) {
        val sharedPref = getSharedPreferences(context)

        Log.d(
            "OnboardingPrefs",
            "Can Show Introduction: " + sharedPref.getBoolean(
                context.resources.getString(R.string.tutorialTracking_canShowIntroduction),
                false
            )
        )
    }

    /**
     *
     */
    fun printFromVariables() {
        Log.d(
            "GlobalPreferencesVars",
            "; Can Show Introduction: " + canShowIntroduction
        )
    }
}
